test/fixtures/aws-ec2-fixture/src/main/java/fixture/aws/ec2/AwsEc2HttpHandler.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+package fixture.aws.ec2;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+
+import org.apache.http.client.utils.URLEncodedUtils;
+import org.elasticsearch.ExceptionsHelper;
+import org.elasticsearch.core.SuppressForbidden;
+import org.elasticsearch.rest.RestStatus;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.BiPredicate;
+import java.util.function.Supplier;
+
+import javax.xml.XMLConstants;
+import javax.xml.stream.XMLOutputFactory;
+import javax.xml.stream.XMLStreamWriter;
+
+import static fixture.aws.AwsCredentialsUtils.checkAuthorization;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.elasticsearch.test.ESTestCase.randomIdentifier;
+import static org.junit.Assert.assertNull;
+
+/**
+ * Minimal HTTP handler that emulates the AWS EC2 DescribeInstances endpoint
+ */
+@SuppressForbidden(reason = "this test uses a HttpServer to emulate the AWS EC2 (DescribeInstances) endpoint")
+public class AwsEc2HttpHandler implements HttpHandler {
```

Comment:
I wonder whether this class and the associated fixture class `AwsEc2HttpFixture` should be more specifically named so that it is clear that they only handle DescribeInstances? I initially thought the two classes would be a 1:1 migration of the deleted `AmazonEC2Fixture`. I later realised the IMDS part has already been [extracted](https://github.com/elastic/elasticsearch/pull/117324).

Reply:
The deleted `AmazonEC2Fixture` was in fact doing a lot more than just serving the EC2 APIs, and that was inappropriate and kinda confusing. This is now just handling the EC2 endpoint alone, i.e. something like `ec2.us-east-1.amazonaws.com`, so I think this is a good name. The fact that it only implements `DescribeInstances` today is incidental, if we needed another thing under the EC2 endpoint then it'd go here too.